internal/pkg/aws/ec2/ec2.go

```diff
@@ -0,0 +1,125 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package ec2 provides a client to make API requests to Amazon Elastic Compute Cloud.
+package ec2
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+)
+
+var (
+	fmtFilterTagApp = fmt.Sprintf("tag:%s", deploy.AppTagKey)
+	fmtFilterTagEnv = fmt.Sprintf("tag:%s", deploy.EnvTagKey)
+	defaultFilter   = ec2.Filter{
+		Name:   aws.String(filterDefault),
+		Values: aws.StringSlice([]string{"true"}),
+	}
+)
+
+const (
+	filterDefault = "default-for-az"
+)
+
+type api interface {
+	DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error)
+	DescribeSecurityGroups(*ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error)
+}
+
+type EC2 struct {
+	client api
+}
+
+func New(s *session.Session) *EC2 {
+	return &EC2{
+		client: ec2.New(s),
+	}
+}
+
+// GetSubnetIDs finds the subnet IDs associated with the environment of the application; if env is a None env,
+// it finds the default subnet IDs.
+func (c *EC2) GetSubnetIDs(app string, env string) ([]string, error) {
```

Comment:
We should refrain from exposing CLI concepts as inputs in our `internal/aws/*` packages (none of our other packages take an app or env for example). Instead, the caller should make the necessary transformation for the filter.

The `internal/aws/*` layer is meant to be just a wrapper around the AWS SDK dependency to make the SDK's functionality more convenient for clients (see `codepipeline` or `cloudformation` as good examples).

Instead, we can expose methods like this:
```suggestion
func (c *EC2) GetSubnetIDs(filters map[string][]string) ([]string, error) {
```

Hope this makes sense!